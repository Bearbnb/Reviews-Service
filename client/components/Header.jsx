import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import Search from './Search.jsx';
import styles from '../../styles/ReviewSummary.css';


const Header = ({ ratingToStars, average, length, onClick, searched }) => {
  const starArr = ratingToStars(average);
  const stars = starArr.map((star) => {
    if (star === 'whole') {
      return (<i id={styles.star} className="fas fa-star" />);
    } if (star === 'half') {
      return (<i id={styles.star} className="fas fa-star-half-alt" />);
    }
    return (<i id={styles.star} className="far fa-star" />);
  });
  return (
    <div className={styles.averageHeader}>
      <div className={styles.textHeader}>
        <div className={styles.reviewTitle}>
          <h2>{length} Reviews </h2>
        </div>
        <div className={styles.headerStars}>
          {stars}
        </div>
      </div>
      <div className={styles.search}>
        <Search onClick={onClick} searched={searched} />
      </div>
    </div>);
};

Header.propTypes = {
  ratingToStars: PropTypes.func.isRequired,
  average: PropTypes.any.isRequired,
  length: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  searched: PropTypes.bool.isRequired,
};

export default Header;
