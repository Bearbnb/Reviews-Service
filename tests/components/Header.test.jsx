import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Header from '../../client/components/Header';

const ratingToStars = (number) => {
  const stars = [];
  let num = number;
  for (let i = 0; i < 5; i += 1) {
    if (num >= 1) {
      stars.push('whole');
    } else if (num > 0) {
      stars.push('half');
    } else {
      stars.push('empty');
    }
    num -= 1;
  }
  return stars;
};
describe('<Header />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Header
      searched
      average={0}
      ratingToStars={ratingToStars}
      length={0}
      term=""
      onClick={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
