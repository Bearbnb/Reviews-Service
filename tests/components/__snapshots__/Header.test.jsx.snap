// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly 1`] = `
<div
  className="averageHeader"
>
  <div
    className="textHeader"
  >
    <div
      className="reviewTitle"
    >
      <h2>
        0
         Reviews 
      </h2>
    </div>
    <div
      className="headerStars"
    >
      <i
        className="far fa-star"
        id="star"
      />
      <i
        className="far fa-star"
        id="star"
      />
      <i
        className="far fa-star"
        id="star"
      />
      <i
        className="far fa-star"
        id="star"
      />
      <i
        className="far fa-star"
        id="star"
      />
    </div>
  </div>
  <div
    className="search"
  >
    <Search
      onClick={[MockFunction]}
      searched={true}
    />
  </div>
</div>
`;
